### PR TITLE
DTA Android now uses Kotlin

### PR DESCRIPTION
**Pull request by toddsouthenbentley**
- Converted the java code to Kotlin
- Added functionality so the Briefcase button will now prompt the user to select a file to open

### CODE REVIEW

**Comment by tcobbs-bentley** [comment]
Since I doubt that others have any knowledge of Kotlin, I will review this and raise any issues I find, and report when I feel it is good. One change Todd made was to add @iTwin/mobilesdkadmins as a CODEOWNER for dta/android, so in the future I will be a valid approver for such changes.

File: test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/FileHelper.kt
Comment (tcobbs-bentley) [inline test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/FileHelper.kt:33]: If close is needed for "not found", it seems like it should also be needed before the return on line 30. If it's not needed before that return, it seems like it shouldn't be needed here either.

**Comment by aruniverse** [comment]
> Since I doubt that others have any knowledge of Kotlin

@simihartstein plz review

File: test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt
Comment (tcobbs-bentley) [inline test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt:93]: I would remove the type for the variable on the left of the equals sign:

```kotlin
        val frontend = object : MobileFrontend(host, "&standalone=true") {
```

**Comment by tcobbs-bentley** [comment]
I made a couple of suggestions, but have otherwise finished my review.

File: test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/FileHelper.kt
Comment (toddsouthenbentley) [reply test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/FileHelper.kt]: I completely reworked the functions as extension functions and moved them into MainActivity. Calling `close` is always required, Kotlin has a nice generic top-level `use` function that calls `close` for you in a finally block.

File: test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt
Comment (toddsouthenbentley) [reply test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt]: Fixed

**Review by tcobbs-bentley — APPROVED** [review]

**Review by pmconne — APPROVED** [review]
@tcobbs-bentley approved. Requeued failed jobs.

**Comment by aruniverse** [comment]
I know @simihartstein has some kotlin experience so hoping he can review when he gets a chance.

**Comment by tcobbs-bentley** [comment]
> I know @simihartstein has some kotlin experience so hoping he can review when he gets a chance.

You can wait for his review, but I believe that Todd and I both have more Kotlin experience than he does. (As far as I know his first Kotlin experience was working on iTwin Mobile SDK. I could be wrong about that, though.)

**Review by tcobbs-bentley — APPROVED** [review]

File: test-apps/display-test-app/android/README.md
Comment (simihartstein) [inline test-apps/display-test-app/android/README.md:26]: should be "opened from there"

**Review by simihartstein — COMMENTED** [review]
Sorry I was late to get to this. Looks good to me aside from a spelling error.

File: test-apps/display-test-app/android/README.md
Comment (toddsouthenbentley) [reply test-apps/display-test-app/android/README.md:26]: Thanks, will fix in a future PR

**Pull request merged**
